### DISCUSSION TITLE
ARM support for Jetson AGX

### DISCUSSION THREAD

**Discussion by mxpfeng** [google/or-tools#3839] [Build and configuration issues]
**What language and solver does this apply to?**
 C++

**Describe the problem you are trying to solve.**
I want to use it on arm(eg,Jetson AGX Xavier), don't know how to do it.
Thanks.

**Comment by Mizux** [comment]
While it is not always tested (take few hours on github runner to run the job see https://github.com/google/or-tools/actions/workflows/aarch64_toolchain.yml) BUT we provide a way to cross-build for arm64.
-> i.e. auto create a arm64 cmake cross toolchain (using the bootlin toolchain) and use the CMake based build. You should run to see usage and you could skim the code:
```sh
./tools/cross_compile.sh -h
```
src: https://github.com/google/or-tools/blob/main/tools/cross_compile.sh#L190

An other solution is to build on a docker with linux/arm64 platform (see docker buildx)
ref: 
* https://github.com/docker-library/official-images#architectures-other-than-amd64
* https://docs.docker.com/build/building/multi-platform/#building-multi-platform-images

Note: Seems Nvidia may provide a toolchain in its SDK
https://forums.developer.nvidia.com/t/jetson-agx-xavier-cross-compile-on-ubuntu/235271/7

**Comment by lperron** [comment]
Moving to discussion. We do not officially support custom arm builds.
